frontend/public/components/monitoring/targets.tsx

```diff
@@ -157,8 +161,9 @@ const Details = withRouter<DetailsProps>(({ loaded, loadError, match, targets })
   const scrapeUrl = atob(match?.params?.scrapeUrl ?? '');
   const target: Target = scrapeUrl ? _.find(targets, { scrapeUrl }) : undefined;
```

Comment:
is there any case scrapeUrl can be `undefined` or `null`?

Reply:
Thanks, I refactored this to hopefully be clearer.

I also realized that it was possible to cause an uncaught error by changing the `scrapeUrl` parameter in the URL to something that `atob` cannot decode, so I've added a `try` / `catch` here too.